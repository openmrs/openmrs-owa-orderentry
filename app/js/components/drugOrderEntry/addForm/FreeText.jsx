import React from 'react';
import PropTypes from 'prop-types';

const FreeText = ({
  fieldErrors,
  fields,
  options,
  handleChange,
  handleSubmit,
  handleCancel,
  handleValidation,
  activateSaveButton,
}) => {
  const { drugInstructions, dispensingQuantity, dispensingUnit } = fields;
  return (
    <div>
      <form className="simple-form-ui">
        <div className="flex-row">
          <div>
            <label name="notes">Complete Instructions</label>
            <textarea
              rows="4"
              cols="80"
              id="notes"
              name="drugInstructions"
              value={drugInstructions}
              onChange={handleChange}
            />
          </div>
        </div>
        <div className="flex-row">
          <div>
            <p className="left label-margin">
              <label>Dispense:</label>
            </p>
            <p className="left p-margin">
              <input
                className={`medium-input ${fieldErrors.dispensingQuantity && "illegalValue"}`}
                placeholder="Quantity"
                name="dispensingQuantity"
                id="dispenseQuantity"
                type="number"
                min="0"
                onBlur={handleValidation}
                onChange={handleChange}
                value={fields.dispensingQuantity} />
              {
                fieldErrors.dispensingQuantity ?
                  <span className="field-error">Required</span>
                  : ""
              }
            </p>
            <p className="left">
              <input
                className={(fields.dispensingQuantity && !fields.dispensingUnit) ? "illegalValue" : ""}
                placeholder="Units"
                name="dispensingUnit"
                id="drugDispensingUnits"
                list="dispensingUnits"
                size="20"
                value={fields.dispensingUnit}
                onBlur={handleValidation}
                onChange={handleChange} />
              <datalist id="dispensingUnits" >
                {
                  options.dispensingUnit &&
                    options.dispensingUnit.map(unit =>
                      (
                        <option key={unit.uuid} value={unit.display} />
                      ))
                }
              </datalist>
              {
                (fields.dispensingQuantity && !fields.dispensingUnit) &&
                  <span className="field-error">Required</span>
              }
            </p>
          </div>
        </div>
        <div className="flex-row">
          <div>
            <button
              type="button"
              className="cancel"
              onClick={handleCancel}
            >
            Cancel
            </button>
          </div>
          <div className="column-1">
            <button
              type="button"
              className="confirm"
              onClick={handleSubmit}
              disabled={activateSaveButton()}
            >
            Add
            </button>
          </div>
        </div>
      </form>
    </div>
  );
};

FreeText.propTypes = {
  fieldErrors: PropTypes.object.isRequired,
  handleValidation: PropTypes.func.isRequired,
  activateSaveButton: PropTypes.func.isRequired,
  fields: PropTypes.object.isRequired,
  options: PropTypes.object,
  handleCancel: PropTypes.func.isRequired,
  handleChange: PropTypes.func.isRequired,
  handleSubmit: PropTypes.func.isRequired,
};

FreeText.defaultProps = {
  options: {},
};

export default FreeText;
