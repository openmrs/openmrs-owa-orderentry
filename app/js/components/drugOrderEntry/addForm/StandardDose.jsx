import React from 'react';
import PropTypes from 'prop-types';
import '../../../../css/grid.scss';

const StandardDose = ({
  fields,
  fieldErrors,
  options,
  handleChange,
  handleValidation,
  activateSaveButton,
  handleSubmit,
  handleCancel,
}) => {
  const {
    dosingUnit, frequency, route, durationUnit, dispensingUnit,
  } = options;
  return (
    <div>
      <form className="simple-form-ui">
        <div className="flex-row">
          <div>
            <p className="left p-margin">
              <input
                className={`small-input ${fieldErrors.dose ? "illegalValue" : ""}`}
                placeholder="Dose"
                id="dose"
                name="dose"
                type="number"
                min="0"
                onBlur={handleValidation}
                onChange={handleChange}
                value={fields.dose}
                required
              />
              {
                fieldErrors.dose ?
                  <span className="field-error">Required</span>
                  : ""
              }
            </p>
            <p className="left p-margin">
              <input
                className={fields.dose && !fields.dosingUnit ? "illegalValue" : ""}
                placeholder="Units"
                id="drugDosingUnits"
                name="dosingUnit"
                list="dosingUnits"
                size="20"
                value={fields.dosingUnit}
                onBlur={handleValidation}
                onChange={handleChange}
                required />
              <datalist id="dosingUnits" >
                {
                  dosingUnit && dosingUnit.map(unit => (
                    <option
                      key={unit.uuid}
                      value={unit.display}
                    />))
                }
              </datalist>
              {
                (fields.dose && !fields.dosingUnit) ?
                  <span className="field-error">Required</span>
                  : ""
              }
            </p>
            <p className="left p-margin">
              <input
                className={fieldErrors.frequency ? "illegalValue" : ""}
                placeholder="Frequency"
                id="frequency"
                name="frequency"
                list="orderFrequencies"
                size="40"
                value={fields.frequency}
                onBlur={handleValidation}
                onChange={handleChange} />
              <datalist id="orderFrequencies" >
                {
                  frequency && frequency.map(frequencyOption => (
                    <option
                      key={frequencyOption.uuid}
                      value={frequencyOption.display}
                    />
                  ))
                }
              </datalist>
              {
                fieldErrors.frequency ?
                  <span className="field-error">Required</span>
                  : ""
              }
            </p>
            <p className="left p-margin">
              <input
                className={fieldErrors.route ? "illegalValue" : ""}
                placeholder="Route"
                id="route"
                name="route"
                list="routes"
                size="20"
                value={fields.route}
                onBlur={handleValidation}
                onChange={handleChange} />
              <datalist id="routes" >
                {
                  route && route.map(routeOption => (
                    <option
                      key={routeOption.uuid}
                      value={routeOption.display}
                    />))
                }
              </datalist>
              {
                fieldErrors.route ?
                  <span className="field-error">Required</span>
                  : ""
              }
            </p>
          </div>
        </div>
        <div className="flex-row">
          <div>
            <p className="left label-margin">
              <label name="reason">As needed for</label>
            </p>
            <p className="left">
              <input
                name="reason"
                placeholder="reason(optional)"
                id="reason"
                type="text"
                size="35"
                onChange={handleChange}
                value={fields.reason} />
            </p>
          </div>
        </div>
        <div className="flex-row">
          <div>
            <p className="left label-margin">
              <label>For</label>
            </p>
            <p className="left p-margin">
              <input
                className="medium-input"
                placeholder="Duration"
                id="duration"
                name="duration"
                type="number"
                min="0"
                onChange={handleChange}
                value={fields.duration} />
            </p>
            <p className="left label-margin">
              <input
                className={fields.duration && !fields.durationUnit ? "illegalValue" : ""}
                placeholder="Units"
                name="durationUnit"
                id="drugDurationUnits"
                list="durationUnits"
                size="20"
                value={fields.durationUnit}
                onBlur={handleValidation}
                onChange={handleChange} />
              <datalist id="durationUnits" >
                {
                  durationUnit && durationUnit.map(unit => (
                    <option
                      key={unit.uuid}
                      value={unit.display}
                    />
                  ))
                }
              </datalist>
              {
                (fields.duration && !fields.durationUnit) ?
                  <span className="field-error">Required</span>
                  : ""
              }
            </p>
            <p className="left">
              <label>total</label>
            </p>
          </div>
        </div>
        <div className="flex-row">
          <div>
            <textarea
              rows="2"
              cols="60"
              id="notes"
              placeholder="Additional instruction not covered above"
              name="drugInstructions"
              onChange={handleChange}
              value={fields.drugInstructions} />
          </div>
        </div>
        <div className="flex-row">
          <div>
            <p className="left label-margin">
              <label>Dispense:</label>
            </p>
            <p className="left p-margin">
              <input
                className={`medium-input ${fieldErrors.dispensingQuantity ? "illegalValue" : ""}`}
                placeholder="Quantity"
                name="dispensingQuantity"
                id="dispenseQuantity"
                type="number"
                min="0"
                onBlur={handleValidation}
                onChange={handleChange}
                value={fields.dispensingQuantity} />
              {
                fieldErrors.dispensingQuantity ?
                  <span className="field-error">Required</span>
                  : ""
              }
            </p>
            <p className="left">
              <input
                className={fields.dispensingQuantity && !fields.dispensingUnit ? "illegalValue" : ""}
                placeholder="Units"
                name="dispensingUnit"
                id="drugDispensingUnits"
                list="dispensingUnits"
                size="20"
                value={fields.dispensingUnit}
                onBlur={handleValidation}
                onChange={handleChange} />
              <datalist id="dispensingUnits" >
                {
                  dispensingUnit && dispensingUnit.map(unit => (
                    <option
                      key={unit.uuid}
                      value={unit.display}
                    />))
                }
              </datalist>
              {
                (fields.dispensingQuantity && !fields.dispensingUnit) ?
                  <span className="field-error">Required</span>
                  : ""
              }
            </p>
          </div>
        </div>
        <div className="flex-row">
          <div>
            <button
              type="button"
              className="cancel"
              onClick={handleCancel}
            >
              Cancel
            </button>
          </div>
          <div className="-1">
            <button
              type="button"
              className="confirm"
              onClick={handleSubmit}
              disabled={activateSaveButton()}>
              Add
            </button>
          </div>
        </div>
      </form>
    </div>
  );
};

StandardDose.propTypes = {
  fields: PropTypes.object.isRequired,
  fieldErrors: PropTypes.object.isRequired,
  options: PropTypes.object.isRequired,
  handleChange: PropTypes.func.isRequired,
  handleValidation: PropTypes.func.isRequired,
  activateSaveButton: PropTypes.func.isRequired,
  handleSubmit: PropTypes.func.isRequired,
  handleCancel: PropTypes.func.isRequired,
};

export default StandardDose;
