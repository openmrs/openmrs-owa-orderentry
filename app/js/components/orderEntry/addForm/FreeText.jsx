import React from 'react';
import PropTypes from 'prop-types';

const FreeText = ({
  fields,
  allConfigurations,
  handleChange,
  handleSubmit,
}) => {
  const { completeInstructions, dispensingQuantity, dispensingUnit } = fields;
  return (
    <div>
      <form>
        <div>
          <div>
            <label name="instructions">Complete Instructions</label>
            <textarea
              id="instructions"
              name="completeInstructions"
              value={completeInstructions}
              onChange={handleChange}
              required
            />
          </div>
          <div>
            <div>
              <label>Dispense:</label>
            </div>
            <div>
              <label name="Quantity">Dose</label>
              <input
                type="number"
                id="Quantity"
                name="dispensingQuantity"
                min="0"
                value={dispensingQuantity}
                onChange={handleChange}
                required
              />
            </div>
            <div>
              <label name="dispenseUnits">Units</label>
              <input
                type="search"
                id="dispenseUnits"
                name="dispensingUnit"
                list="units"
                size="10"
                value={dispensingUnit}
                onChange={handleChange}
                required />
              <datalist id="units" >
                {
                  allConfigurations &&
                  (
                    allConfigurations.drugDispensingUnits &&
                  allConfigurations.drugDispensingUnits.map(unit =>
                    (
                      <option key={unit.uuid} value={unit.display} />
                    ))
                  )
                }
              </datalist>
            </div>
          </div>
          <div>
            <input type="button" className="button cancel" value="Cancel" />
            <input type="button" onClick={handleSubmit} className="button confirm right" value="Save" />
          </div>
        </div>
      </form>
    </div>
  );
};

FreeText.propTypes = {
  fields: PropTypes.object.isRequired,
  allConfigurations: PropTypes.shape({
    drugDispensingUnits: PropTypes.array,
  }),
  handleChange: PropTypes.func.isRequired,
  handleSubmit: PropTypes.func.isRequired,
};

FreeText.defaultProps = {
  allConfigurations: {},
};

export default FreeText;
