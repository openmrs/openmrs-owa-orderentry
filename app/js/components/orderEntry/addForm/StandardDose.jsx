import React from 'react';
import PropTypes from 'prop-types';
import '../../../../css/grid.scss';

const StandardDose = ({
  fields,
  fieldErrors,
  careSetting,
  allConfigurations,
  handleChange,
  handleValidation,
  activateSaveButton,
  handleSubmit,
  handleCancel,
}) => {
  const {
    drugDosingUnits, orderFrequencies, drugRoutes, durationUnits, drugDispensingUnits,
  } = allConfigurations;
  return (
    <div>
      <form className="simple-form-ui">
        <div className="grid-row">
          <div className="column-10">
            <p className="left p-margin">
              <input
                className={`small-input ${fieldErrors.dose ? "illegalValue" : ""}`}
                placeholder="Dose"
                id="dose"
                name="dose"
                type="number"
                min="0"
                onBlur={handleValidation}
                onChange={handleChange}
                value={fields.dose}
                required
              />
              {
                fieldErrors.dose ?
                  <span className="field-error">Required</span>
                  : ""
              }
            </p>
            <p className="left p-margin">
              <input
                className={fields.dose && !fields.dosingUnit ? "illegalValue" : ""}
                placeholder="Units"
                id="drugDosingUnits"
                name="dosingUnit"
                list="dosingUnits"
                size="7"
                value={fields.dosingUnit}
                onBlur={handleValidation}
                onChange={handleChange}
                required />
              <datalist id="dosingUnits" >
                {
                  drugDosingUnits
                && drugDosingUnits.map(unit => (
                  <option
                    key={unit.uuid}
                    value={unit.display}
                  />))
                }
              </datalist>
              {
                (fields.dose && !fields.dosingUnit) ?
                  <span className="field-error">Required</span>
                  : ""
              }
            </p>
            <p className="left p-margin">
              <input
                className={fieldErrors.frequency ? "illegalValue" : ""}
                placeholder="Frequency"
                id="frequency"
                name="frequency"
                list="orderFrequencies"
                size="20"
                value={fields.frequency}
                onBlur={handleValidation}
                onChange={handleChange} />
              <datalist id="orderFrequencies" >
                {
                  orderFrequencies &&
                orderFrequencies.map(frequency => (
                  <option
                    key={frequency.uuid}
                    value={frequency.display}
                  />
                ))
                }
              </datalist>
              {
                fieldErrors.frequency ?
                  <span className="field-error">Required</span>
                  : ""
              }
            </p>
            <p className="left p-margin">
              <input
                className={fieldErrors.route ? "illegalValue" : ""}
                placeholder="Route"
                id="route"
                name="route"
                list="routes"
                size="20"
                value={fields.route}
                onBlur={handleValidation}
                onChange={handleChange} />
              <datalist id="routes" >
                {
                  drugRoutes &&
                drugRoutes.map(route => (
                  <option
                    key={route.uuid}
                    value={route.display}
                  />))
                }
              </datalist>
              {
                fieldErrors.route ?
                  <span className="field-error">Required</span>
                  : ""
              }
            </p>
          </div>
        </div>
        <div className="grid-row">
          <div className="column-7">
            <p className="left label-margin">
              <label name="reason">As needed for</label>
            </p>
            <p className="left">
              <input
                name="reason"
                placeholder="reason(optional)"
                id="reason"
                type="text"
                size="30"
                onChange={handleChange}
                value={fields.reason} />
            </p>
          </div>
        </div>
        <div className="grid-row">
          <div className="column-7">
            <p className="left label-margin">
              <label>For</label>
            </p>
            <p className="left p-margin">
              <input
                className="medium-input"
                placeholder="Duration"
                id="duration"
                name="duration"
                type="number"
                min="0"
                onChange={handleChange}
                value={fields.duration} />
            </p>
            <p className="left label-margin">
              <input
                className={fields.duration && !fields.durationUnit ? "illegalValue" : ""}
                placeholder="Units"
                name="durationUnit"
                id="drugDurationUnits"
                list="durationUnits"
                size="8"
                value={fields.durationUnit}
                onBlur={handleValidation}
                onChange={handleChange} />
              <datalist id="durationUnits" >
                {
                  durationUnits &&
                durationUnits.map(unit => (
                  <option
                    key={unit.uuid}
                    value={unit.display}
                  />
                ))
                }
              </datalist>
              {
                (fields.duration && !fields.durationUnit) ?
                  <span className="field-error">Required</span>
                  : ""
              }
            </p>
            <p className="left">
              <label>total</label>
            </p>
          </div>
        </div>
        <div className="grid-row">
          <div className="column-6">
            <textarea
              rows="2"
              cols="60"
              id="notes"
              placeholder="Additional instruction not covered above"
              name="drugInstructions"
              onChange={handleChange}
              value={fields.drugInstructions} />
          </div>
        </div>
        { careSetting.display === 'Outpatient' ?
          <div className="grid-row">
            <div className="column-7">
              <label>For outpatient orders </label>
              <p className="left label-margin">
                <label>Dispense:</label>
              </p>
              <p className="left p-margin">
                <input
                  className={`medium-input ${fieldErrors.dispensingQuantity ? "illegalValue" : ""}`}
                  placeholder="Quantity"
                  name="dispensingQuantity"
                  id="dispenseQuantity"
                  type="number"
                  min="0"
                  onBlur={handleValidation}
                  onChange={handleChange}
                  value={fields.dispensingQuantity} />
                {
                  fieldErrors.dispensingQuantity ?
                    <span className="field-error">Required</span>
                    : ""
                }
              </p>
              <p className="left">
                <input
                  className={fields.dispensingQuantity && !fields.dispensingUnit ? "illegalValue" : ""}
                  placeholder="Units"
                  name="dispensingUnit"
                  id="drugDispensingUnits"
                  list="dispensingUnits"
                  size="8"
                  value={fields.dispensingUnit}
                  onBlur={handleValidation}
                  onChange={handleChange} />
                <datalist id="dispensingUnits" >
                  {
                    drugDispensingUnits &&
                  drugDispensingUnits.map(unit => (
                    <option
                      key={unit.uuid}
                      value={unit.display}
                    />))
                  }
                </datalist>
                {
                  (fields.dispensingQuantity && !fields.dispensingUnit) ?
                    <span className="field-error">Required</span>
                    : ""
                }
              </p>
            </div>
          </div> :
          <br />
        }
        <div className="grid-row">
          <div className="column-1">
            <button
              type="button"
              className="cancel"
              onClick={handleCancel}
            >
              Cancel
            </button>
          </div>
          <div className="column-1 pull-right-8">
            <button
              type="button"
              className="confirm"
              onClick={handleSubmit}
              disabled={activateSaveButton()}>
                Save
            </button>
          </div>
        </div>
      </form>
    </div>
  );
};

StandardDose.propTypes = {
  fields: PropTypes.object.isRequired,
  fieldErrors: PropTypes.object.isRequired,
  careSetting: PropTypes.object.isRequired,
  allConfigurations: PropTypes.object.isRequired,
  handleChange: PropTypes.func.isRequired,
  handleValidation: PropTypes.func.isRequired,
  activateSaveButton: PropTypes.func.isRequired,
  handleSubmit: PropTypes.func.isRequired,
  handleCancel: PropTypes.func.isRequired,
};

export default StandardDose;
