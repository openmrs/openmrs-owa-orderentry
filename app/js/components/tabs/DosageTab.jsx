import React from 'react';
import PropTypes from 'prop-types';

const DosageTab = props => (
  <a
    role="presentation"
    onClick={(event) => {
      event.preventDefault();
      props.onClick(props.tabIndex, props.tabName);
    }}>
    <span className={props.icon} />
    {props.tabName}
  </a>
);

DosageTab.propTypes = {
  icon: PropTypes.string.isRequired,
  tabName: PropTypes.string.isRequired,
};

export default DosageTab;
