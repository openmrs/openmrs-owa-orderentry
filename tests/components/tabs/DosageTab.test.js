import React from 'react';
import DosageTab from '../../../app/js/components/tabs/DosageTab'
import StandardDose from '../../../app/js/components/drugOrderEntry/addForm/StandardDose'
import FreeText from '../../../app/js/components/drugOrderEntry/addForm/FreeText'

describe('Dosage Tab component', () => {
    const props = {
      activeTabIndex:0
    }
    const dosageTabProps = {
      icon: '',
      tabName: '',
      onClick: jest.fn(),
      tabIndex: 1,
    };
    const standardDoseProps = {
      fields: {},
      fieldErrors: {},
      careSetting: {},
      options: {},
      handleChange: jest.fn(),
      handleValidation: jest.fn(),
      activateSaveButton: jest.fn(),
      handleSubmit: jest.fn(),
      handleCancel: jest.fn(),
    };
    const freeTextProps = {
      fieldErrors: {},
      handleValidation: jest.fn(),
      activateSaveButton: jest.fn(),
      fields: {},
      options: {},
      handleCancel: jest.fn(),
      handleChange: jest.fn(),
      handleSubmit: jest.fn(),
      careSetting: {
        display: '',
      },
    };
    it('should render without crashing', () => {
        const wrapper = shallow(<DosageTab {...dosageTabProps}/>);
        expect(wrapper).toMatchSnapshot();
    });
    it('should render the standard form', () => {
      const wrapper = shallow(<DosageTab {...dosageTabProps}><StandardDose {...standardDoseProps}/></DosageTab>);
      expect(wrapper).toMatchSnapshot();
    });
    it('should render the free text form', () => {
      const wrapper = shallow(<DosageTab {...dosageTabProps}><FreeText {...freeTextProps}/></DosageTab>);
      expect(wrapper).toMatchSnapshot();
  });
});

describe('Test click events', () => {
  const props = {
    icon: '',
    tabName: '',
    onClick: jest.fn(),
    tabIndex: 1,
  };
  it('should trigger event when clicked', () => {
    const wrapper = shallow(<DosageTab {...props}/>);
    expect(props.onClick.mock.calls.length).toBe(0);
    wrapper.find('a').simulate('click', {
        preventDefault: () => {},
      });
    expect(props.onClick.mock.calls.length).toBe(1);
  });
});
