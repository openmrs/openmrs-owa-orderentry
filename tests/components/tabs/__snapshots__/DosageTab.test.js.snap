// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dosage Tab component should render the free text form 1`] = `
<a
  onClick={[Function]}
  role="presentation"
>
  <span
    className=""
  />
</a>
`;

exports[`Dosage Tab component should render the standard form 1`] = `
<a
  onClick={[Function]}
  role="presentation"
>
  <span
    className=""
  />
</a>
`;

exports[`Dosage Tab component should render without crashing 1`] = `
<a
  onClick={[Function]}
  role="presentation"
>
  <span
    className=""
  />
</a>
`;
